### PR TITLE
[MRG+2] DOC changed documentation for axvspan to numpydoc format

### PR DESCRIPTION
**Pull request by NelleV**
Updated documentation of axvspan function to numpydoc format, as part of a project during the ASPP2016 Summer School

supersedes #7039 (fixes the pep8 compliancy issues).

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -862,39 +862,50 @@ def axvspan(self, xmin, xmax, ymin=0, ymax=1, **kwargs):
...
+            Optional parameters are properties of the class
+            matplotlib.patches.Polygon.
 
-        Valid kwargs are :class:`~matplotlib.patches.Polygon`
-        properties:
+        See Also
+        --------
+        axhspan
 
-        %(Polygon)s
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: We're still removing this here?

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -862,39 +862,50 @@ def axvspan(self, xmin, xmax, ymin=0, ymax=1, **kwargs):
...
+        -------
+        rectangle : matplotlib.patches.Polygon
+            Vertical span (rectangle) from (xmin, ymin) to (xmax, ymax).
 
-            >>> axvspan(1.25, 1.55, facecolor='g', alpha=0.5)
+        Other Parameters
+        ----------------
+        **kwargs
+            Optional parameters are properties of the class
+            matplotlib.patches.Polygon.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Should link?

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: I think this is the general consensus to move away from these, and have nicer documentation on the artists themselves. At least that's what I understood from different conversation, even during the matplotlib call or on tickets.

Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: I may document this in a MEP somewhere.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -862,39 +862,50 @@ def axvspan(self, xmin, xmax, ymin=0, ymax=1, **kwargs):
...
+        axhspan
 
-        %(Polygon)s
+        Examples
+        --------
+        Draw a vertical, green, translucent rectangle from x = 1.25 to
+        x = 1.55 that spans the yrange of the axes.
 
-        .. seealso::
+        >>> axvspan(1.25, 1.55, facecolor='g', alpha=0.5)
```
Comment (phobson) [inline lib/matplotlib/axes/_axes.py]: What's the consensus on the style and  "completeness" of examples like this? 

My thought is that something like:

```python
import matplotlib.pyplot as plt
fig, ax  = plt.subplots()
ax.axvspan(1.25, 1.55, facecolor='g', alpha=0.5)
plt.show()
```

or a link to the gallery/examples would be better.

I don't feel very strongly about that, though.

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -862,39 +862,50 @@ def axvspan(self, xmin, xmax, ymin=0, ymax=1, **kwargs):
...
 
-            :meth:`axhspan`
-                for example plot and source code
```
Comment (phobson) [inline lib/matplotlib/axes/_axes.py]: I think the `See Also` sections are helpful. But I'd understand if they're a pain to maintain.

File: lib/matplotlib/axes/_axes.py (hunk 3 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: I agree that a full runnable example would be much better. I am not a big fan of links in the gallery as they are only good for sphinx rendered documentation, but do not render properly in terminal docstring.

File: lib/matplotlib/axes/_axes.py (hunk 4 above)
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: It wasn't removed; it just got a numpydoc heading instead of a sphinx one.

Comment (phobson) [inline lib/matplotlib/axes/_axes.py]: Ahh yes. I see that now.

**Comment by phobson** [comment]
backported to v2.x as 05fec775281389b9e27e17df512eb6b5da990bc4

**Pull request merged**